.github/workflows/test.yml

```diff
@@ -0,0 +1,50 @@
+name: Unit Tests
+
+on:
+  push:
+  pull_request:
+  # Allow manually triggering the workflow.
+  workflow_dispatch:
+
+# Cancels all previous workflow runs for the same branch that have not yet completed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
```

Comment:
please add `fail-fast: false`. We don't want to cancel jobs of the matrix when one of them fails (we want to have the tests on all PHP versions being run, allowing us to see whether a failure is limited to some versions or no)

Reply:
Done: please note that I'll be adding quite a few extra test runs in the PR to address the widening of the version support, so that will increase redundancy on failure, but then again, this repo is probably not that active, so it shouldn't be a big resource drain on GH.